tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go

```diff
@@ -0,0 +1,216 @@
+			Expect(err).ToNot(HaveOccurred(), "Failed to compare ZTP version string")
+
+			if !versionInRange {
+				Skip("ZTP Siteconfig operator tests require ZTP 4.17 or later")
+			}
+
+		})
+
+		AfterEach(func() {
+			// Recreate the ClusterInstance custom resource.
+			By("resetting the clusters app back to the original settings")
+			err := gitdetails.SetGitDetailsInArgoCd(
+				tsparams.ArgoCdClustersAppName, tsparams.ArgoCdAppDetails[tsparams.ArgoCdClustersAppName],
+				true, false)
+			Expect(err).ToNot(HaveOccurred(), "Failed to reset clusters app git details")
+
+			// Test teardown expected results validation.
+			By("checking the infra env manifests exists on hub")
+			_, err = assisted.PullInfraEnvInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find spoke infra env manifests")
+
+			By("checking the bare metal host manifests exists on hub")
+			_, err = bmh.Pull(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find spoke bmh manifests")
+
+			By("checking the cluster deployment manifests exists on hub")
+			_, err = hive.PullClusterDeployment(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find spoke cluster deployment manifests")
+
+			By("checking the NM state config manifests exists on hub")
+			nmStateConfigList, err := assisted.ListNmStateConfigs(HubAPIClient, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to list NM state config manifests")
+			Expect(nmStateConfigList).ToNot(BeEmpty(), "Failed to find NM state config manifests")
+
+			By("checking the klusterlet addon config manifests exists on hub")
+			_, err = ocm.PullKAC(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find spoke kac manifests")
+
+			By("checking the agent cluster install manifests exists on hub")
+			_, err = assisted.PullAgentClusterInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to find spoke agent cluster install manifests")
+		})
+
+		// 75374 - Detaching the AI multi-node openshift (MNO) spoke cluster.
+		It("Validate detaching the AI multi-node openshift spoke cluster", reportxml.ID("75374"), func() {
+			By("checking spoke cluster type")
+			spokeClusterType, err := rancluster.CheckSpokeClusterType(RANConfig.Spoke1APIClient)
+			Expect(err).ToNot(HaveOccurred(), "Failed to fetch spoke cluster type")
+
+			if spokeClusterType == ranparam.SNOCluster {
+				Skip("This test only applies to standard or multi-node openshift spoke cluster")
+			}
+
+			// Test step 1-Delete default assisted installer template reference ConfigMap CRs after spoke cluster installed.
+			By("deleting default assisted installer template reference ConfigMap custom resources", func() {
```

Comment:
could you remove the `func() {}` inside this test case too please?

Reply:
done.